From c348e21c5bafa448fc30e3ad81f8f731dca2a00e Mon Sep 17 00:00:00 2001
From: Sughosh Ganu <sughosh.ganu@linaro.org>
Date: Thu, 1 Feb 2024 12:42:40 +0530
Subject: [PATCH 563/954] feat(fwu): document the config flag for including
 image info in the FWU metadata

The version 2 of the FWU metadata structure is designed such that the
information on the updatable images can be omitted from the metadata
structure. Add a config flag, PSA_FWU_METADATA_FW_STORE_DESC, which is
used to select whether the metadata structure has this information
included or not. It's value is set to 1 by default.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git cb59bb80e088e595898a3655f73c87903786124a feat(fwu): document the config flag for including image info in the FWU metadata]
Change-Id: Id6c99455db768edd59b0a316051432a900d30076
Signed-off-by: Sughosh Ganu <sughosh.ganu@linaro.org>
(cherry picked from commit 7ae16196cc73a580f298734bb98f2ccb210e3ba9)
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/372644
Tested-by: Yann GAUTIER <yann.gautier@foss.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
---
 docs/getting_started/build-options.rst | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/docs/getting_started/build-options.rst b/docs/getting_started/build-options.rst
index 06b78e25b8..dd4bf94292 100644
--- a/docs/getting_started/build-options.rst
+++ b/docs/getting_started/build-options.rst
@@ -1332,6 +1332,15 @@ Firmware update options
    This flag is used in defining the firmware update metadata structure. This
    flag is by default set to '1'.
 
+- ``PSA_FWU_METADATA_FW_STORE_DESC``: To be enabled when the FWU
+   metadata contains image description. The default value is 1.
+
+   The version 2 of the FWU metadata allows for an opaque metadata
+   structure where a platform can choose to not include the firmware
+   store description in the metadata structure. This option indicates
+   if the firmware store description, which provides information on
+   the updatable images is part of the structure.
+
 --------------
 
 *Copyright (c) 2019-2025, Arm Limited. All rights reserved.*
-- 
2.53.0

